cmd/fetch.go

```diff
@@ -35,11 +35,26 @@ func fetchCmd() *cobra.Command {
 		Short:  "Fetch SBOM file(s) from HTTP(S), OCI, or Git URLs",
 		Long:   "Fetch SBOM file(s) from HTTP(S), OCI, or Git URLs",
 		Run: func(_ *cobra.Command, _ []string) {
-			var err error
+			var (
+				err    error
+				output *os.File
+			)
+
 			logger = utils.NewLogger("fetch")
 
+			if string(outputFile) != "" {
+				out, err := os.Create(string(outputFile))
+				if err != nil {
+					logger.Fatal("error creating output file", "outputFile", outputFile)
+				}
+
+				output = out
+
+				defer output.Close()
```

Comment:
This one was problematic because of the assignment with `output, err :=` creating a new instance of `output` that goes out of scope after this if block. I guess since both `output` and `err` are declared earlier this could change to `output, err =` without the `:`; I'll give it a shot.

The next PR makes this one a moot point though, instead of passing the `*os.File` as a parameter directly, it adds a `FetchOptions` struct type with an `OutputFile` field and this becomes

```go
opts.OutputFile = out

defer opts.OutputFile.Close()
```

Reply:
Made the change, thanks!